connect/util.go

```diff
@@ -0,0 +1,304 @@
+	if err != nil {
+		return fmt.Errorf("error trying to parse JSON : '%s'", err.Error())
+	}
+
+	csrInfo := types.CSRInfo{}
+	err = json.Unmarshal(response, &csrInfo)
+	if err != nil {
+		return fmt.Errorf("error trying to get CSR Information : '%s'", err.Error())
+	}
+
+	c.CsrInfo = &csrInfo
+	err = c.StorageInterface.WriteData("config.json", response)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	return err
+}
+
+func (c *KymaConnector) PopulateInfo() error {
+
+	resp, err := c.SecureClient.Get(c.CsrInfo.API.InfoUrl)
+
+	if err != nil {
+		return fmt.Errorf("error trying to get info : '%s'", err.Error())
+	}
+	if resp == nil {
+		return fmt.Errorf("did not recieve a response from info URL")
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("recieved non OK status code from info URL")
+	}
+
+	//unmarshal response json and store in csrInfo
+	response, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error trying to parse JSON : '%s'", err.Error())
+	}
+
+	info := types.Info{}
+	err = json.Unmarshal(response, &info)
+	if err != nil {
+		return fmt.Errorf("error trying to get CSR Information : '%s'", err.Error())
+	}
+
+	c.Info = &info
+
+	err = c.StorageInterface.WriteData("info.json", response)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	return err
+}
+
+func (c *KymaConnector) GetCertSigningRequest() error {
+	keys, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	parts := strings.Split(c.CsrInfo.Certificate.Subject, ",")
+
+	var org, orgUnit, location, street, country, appName string
+
+	for i := range parts {
+		subjectTitle := strings.Split(parts[i], "=")
+		switch subjectTitle[0] {
+		case "O":
+			org = subjectTitle[1]
+		case "OU":
+			orgUnit = subjectTitle[1]
+		case "L":
+			location = subjectTitle[1]
+		case "ST":
+			street = subjectTitle[1]
+		case "C":
+			country = subjectTitle[1]
+		case "CN":
+			appName = subjectTitle[1]
+		}
+	}
+
+	pkixName := pkix.Name{
+		Organization:       []string{org},
+		OrganizationalUnit: []string{orgUnit},
+		Locality:           []string{location},
+		StreetAddress:      []string{street},
+		Country:            []string{country},
+		CommonName:         appName,
+		Province:           []string{"Waldorf"},
+	}
+
+	// create CSR
+	var csrTemplate = x509.CertificateRequest{
+		Subject:            pkixName,
+		SignatureAlgorithm: x509.SHA256WithRSA,
+	}
+
+	csrCertificate, err := x509.CreateCertificateRequest(rand.Reader, &csrTemplate, keys)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	csr := pem.EncodeToMemory(&pem.Block{
+		Type: "CERTIFICATE REQUEST", Bytes: csrCertificate,
+	})
+
+	var privateKey bytes.Buffer
+	err = pem.Encode(&privateKey, &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(keys)})
+
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	c.Ca.PrivateKey = privateKey.String()
+	c.Ca.Csr = string(csr)
+	return err
+}
+
```

Comment:
Maybe rename to something like retrieveClientCert (to make clear there is no response from this function except of the error).